lte/gateway/python/magma/pipelined/qos/common.py

```diff
@@ -216,9 +231,15 @@ def callback(fut):
                         subscriber.update_rule(ip_addr, rule_num, d, v)
 
                         qid_state = qos_state[v]
-                        if qid_state['ambr'] != 0:
+                        if qid_state['ambr_qid'] != 0:
                             session = subscriber.get_or_create_session(ip_addr)
-                            session.set_ambr(d, qid_state['ambr'])
+                            ambr_qid = qid_state['ambr_qid']
+                            leaf = 0
+                            # its AMBR QoS handle if its config matches with parent
+                            # pylint: disable=too-many-function-args
+                            if self.impl.same_qos_config(d, ambr_qid, v):
```

Comment:
Minor nit: during recovery, I wonder if we can just read the entire TC class output and figure out if the qos configs are the same rather than doing this for every qid. This might be too slow if we have few 10-100s of queues.

Reply:
I agree but I would like to keep it same since this is recovery process at init time. and I do not want to change entire recovery process to optimize it.

Reply:
Sure. I once ran into this when I had stateless pipelined turned on and checked in TeraVM setup. It took a while last time around. If we aren't turning on stateless pipelined, you can ignore this comment and may be later we can test on how bad it really is and fix it if necessary.

Reply:
are you saying existing QoS implementation has issue in stateless mode due to system command execution?

Reply:
I explored creating the leaf node for APN ambr lazily after first dedicated bearer added, but it got complicated, since I need to change pipelineD metadata and OVS flow for APN ambr. in interest of time. I would like to keep it same for now and explore other ideas to improve it later.